### PR TITLE
Adding CMake build

### PR DESCRIPTION
**Pull request by mijamo911**

### CODE REVIEW

File: ionc/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  ion_writer.c
+  ion_writer_text.c
+)
+set_property(TARGET ionc PROPERTY C_STANDARD 99)
+target_include_directories(ionc PUBLIC
+  .
+  inc/
+
+)
+target_link_libraries(ionc decNumber m)
```
Comment (mijamo911) [inline ionc/CMakeLists.txt]: "m" is the math library, which isn't linked be default for historical reasons

File: ionc/ion_reader_text.c [hunk 2]
```diff
@@ -651,8 +655,6 @@ enum version_marker_result _ion_reader_text_parse_version_marker(ION_STRING* ver
     return SUCCESS;
 }
 
-extern inline int add_digit(int i, char digit);
```
Comment (mijamo911) [inline ionc/ion_reader_text.c]: Switched this guy to inline static as originally intended

File: test/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,10 @@
+add_executable(tester
```
Comment (mijamo911) [inline test/CMakeLists.txt]: tester worked on the first try, hooray! All tests pass

We may want to model this as an actual CMake test, haven't done a whole lot of investigation but it appears we can pass args to an executable so we could just pass in the path to the ion-tests submodule to make it super easy

File: .gitignore [hunk 4]
```diff
@@ -0,0 +1,6 @@
+CMakeCache.txt
+CMakeFiles/
+*/CMakeFiles/
+cmake_install.cmake
```
Comment (almann) [inline .gitignore]: Is this line redundant to the one below it?

File: .gitignore [hunk 5]
```diff
@@ -0,0 +1,6 @@
+CMakeCache.txt
+CMakeFiles/
```
Comment (almann) [inline .gitignore]: Is this line redundant to the one below it?

**Comment by almann** [comment]
Overall looks good--can we streamline the ignore file?

File: .gitignore (hunk 4 above)
Comment (mijamo911) [inline .gitignore]: Not from my testing, both were required for the desired effect

**Pull request merged**